apps/openmw/mwrender/sky.cpp

```diff
@@ -668,7 +671,9 @@ namespace MWRender
                     ps->setUserValue("simpleLighting", true);
                 }
 
-                mSceneManager->recreateShaders(mParticleNode);
+                Shader::ShaderVisitor shaderVisitor(mSceneManager->getShaderVisitorTemplate());
+                mParticleNode->accept(shaderVisitor);
+                // TODO: pass simpleLighting define to shaderVisitor directly
```

Comment:
My plan here is to add a `ShaderVisitor::getDefineMap` we can add to, but we will first need to support optional defines retrieval in our shaders.